sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala

```diff
@@ -308,10 +308,10 @@ package object dsl {
         AttributeReference(s, arrayType)()
 
       /** Creates a new AttributeReference of type map */
-      def map(keyType: DataType, valueType: DataType): AttributeReference =
-        map(MapType(keyType, valueType))
+      def mapAttr(keyType: DataType, valueType: DataType): AttributeReference =
+        mapAttr(MapType(keyType, valueType))
 
-      def map(mapType: MapType): AttributeReference =
+      def mapAttr(mapType: MapType): AttributeReference =
         AttributeReference(s, mapType, nullable = true)()
 
```

Comment:
If you have fixed all the places, shall we remove DSL APIs that take Symbol?

Reply:
I've removed and I found most symbol literals are used with DSL APIs (changed 114 files)...